provider/pkg/provider/provider.go

```diff
@@ -1382,6 +1382,12 @@ func (k *kubeProvider) Diff(
 			// object in a new namespace and then deleting the old one).
 			newInputs.GetNamespace() == oldInputs.GetNamespace()
 
+	// If the replaceOnUpdate annotation is "true", then schedule the resource for replacement.
+	if metadata.ReplaceOnUpdate(newInputs) {
+		hasChanges = pulumirpc.DiffResponse_DIFF_SOME
+		replaces = append(replaces, `.metadata.annotations["pulumi.com/replaceOnUpdate"]`)
```

Comment:
Will this force a diff and replacement *always*, even if there are no changes?  I thought the goal was to replace only if there were changes?

Reply:
> Will this force a diff and replacement always, even if there are no changes?

Correct. I thought this would be required for the Job replacement case, but narrowing it to only apply on changes does seem preferable. I'll do some more testing today.